neuro_evolution_ctrnn/tools/experiment.py

```diff
@@ -46,6 +48,14 @@ def __init__(self, configuration: ExperimentCfg, result_path, processing_framewo
             self.brain_class = LSTMPyTorch
         elif self.config.brain.type == "LSTM_NumPy":
             self.brain_class = LSTMNumPy
+        elif self.config.brain.type == "ELMAN":
+            self.brain_class = ElmanNetwork
+        elif self.config.brain.type == "ElmanTorch":
+            self.brain_class = ElmanPytorch
+        elif self.config.brain.type == "GruTorch":
+            self.brain_class = GruPyTorch
+        elif self.config.brain.type == "GRU":
+            self.brain_class = GruNetwork
```

Comment:
Das ist (wiedereinmal) etwas hässliches, was jedoch nicht deine Schuld ist, sondern meine. Wir hatten ein ähnliches problem schonmal an einer anderen stelle. Im wesentlich ist hier das problem, dass gegen Open–closed principle verstoßen wird, iirc

Ich habe eine Issue dazu angelegt. https://github.com/neuroevolution-ai/NeuroEvolution-CTRNN_new/issues/54

Reply:
ist nur fyi. Du kannst diesen Kommentar gerne direkt auf "resolved" setzen

Reply:
hab's geändert in #55. Das ist an dieser Stelle also erstmal erledigt